### DISCUSSION TITLE
Add flag to automatically select the best match and exit

### DISCUSSION THREAD

**Discussion by DannyBen** [junegunn/fzf#3017] [Ideas]
I am looking for something that behaves like this existing flag:

```
-1, --select-1        Automatically select the only match
```

only this new proposed flag should select the best matching item, even if it is not the only one, perhaps:

```
-1, --select-1        Automatically select the only match
    --select-best     Automatically select the best match
```

**Comment by junegunn** [comment]
```sh
seq 1000 | fzf --filter 55 | head -1
```

**Reply by DannyBen** [reply]
Thanks. Interesting.
Same could be said for `--select-1` no? Yet it exists.

**Reply by junegunn** [reply]
Nope, it serves a different purpose. Optional interactivity.
